### ISSUE TITLE
Convert audfromat.Database.index to a method

### ISSUE THREAD

**Issue by hagenw** [audeering/audformat#200]
A user might be interested to get the index for all files or segments included in one split.

To get all segments involved in the a split called `test` you would need to do the following at the moment:

```python
split = 'test'
type = 'segmented'
tables = [t for t in db.tables if db[t].split_id == split]
tables = [t for t in tables if audformat.index_type(db[t].index) == type]
index = audformat.utils.union([db[t].index for t in tables])
```

We could make this easier by converting the property `audformat.Database.index` into a method:

```python
index = db.index(split='test', index_type='segmented')
```

The only problem is that `Database.index` is a property and we would break the API if we convert it to a method.
If we don't find a way to achieve this, we should maybe not target it or use another name for the proposed method.

**Comment by hagenw** [comment]
This would be a solution. I'm not completely sure if we should limit the functionality already in the name, as you can in principle filter after split, media, index type, but split is most likely the most important attribute.

**Comment by frankenjoe** [comment]
> The only problem is that Database.index is a property and we would break the API

Actually, I just realized that we do not have a property `Database.index` yet, neither an attribute or method.

**Comment by hagenw** [comment]
You are right, we only have it for the single tables. So it should be straightforward to solve this issue.
